### PR TITLE
Credit memo refunds

### PR DESCRIPTION
**Pull request by ccondra**
## Purpose of this pull request

The purpose of this request is to clarify the credit memo refund process for online (credit) and offline (cash/money order) orders. The original topic showed only how to do an offline refund. This rework includes separate instructions for credit purchases (including credit card, store credit, and company credit) and cash/money order purchases.  To complicate matters, the workflow for credit card purchases depends on the Payment Action setting in the payment method configuration. This major rework of this topic addresses all of these possible scenarios.

I updated some screenshots, and added a few new ones. For consistency with our current style guide,  I changed the topic title from "Issuing a Credit Memo "to "Create a Credit Memo".

## Affected documentation pages

https://docs.magento.com/m2/ee/user_guide/sales/credit-memo-create.html

## Affected Magento editions

- [ x ] Open Source
- [ x ] Commerce
- [ x ] B2B

## Additional information

Staging # 414

### CODE REVIEW

File: src/sales/credit-memo-create.md [hunk 1]
```diff
@@ -1,115 +1,179 @@
...
+
+- Purchases made by credit card can be refunded either online or offline.
+- Purchases made by check or money order must be refunded offline.
+
+Any credit memo with a [status]({% link sales/order-status.md %}) of “open” has an outstanding refund due. The refund workflow for orders paid by credit card is determined by the Payment Action setting in the configuration of the payment method. Refunds cannot be issued until the transaction is settled.
+
+- If Payment Action is set to `Authorize`, you must first generate the invoice from the Magento Admin before a credit memo can be created.
+- If Payment Action is set to `Authorize and Capture`, the invoice has already been  generated by the payment processor, but the funds aren't available until the transaction is settled. This brief waiting period is recommended by many payment processors as a security measure, and can usually be handled automatically.Transactions can also be settled manually from your merchant account with the payment processor.
```
Comment (lazyguru) [inline src/sales/credit-memo-create.md]:
```suggestion
- If Payment Action is set to `Authorize and Capture`, the invoice has already been generated by the payment processor, but the funds aren't available until the transaction is settled. This brief waiting period is recommended by many payment processors as a security measure, and can usually be handled automatically. Transactions can also be settled manually from your merchant account with the payment processor.
```

Comment (lazyguru) [reply src/sales/credit-memo-create.md]: A couple of spacing issues here:
- There is an extra space between "been  generated"
- There is no space between "automatically.Transactions"

File: src/sales/credit-memo-create.md [hunk 2]
```diff
@@ -1,115 +1,179 @@
...
 
-       The Return to Stock checkbox appears only if inventory Stock Options are set to `Decrease Stock When Order Is Placed`.
+      The product returns to stock automatically if [Product Stock Options]({% link configuration/catalog/inventory.md %}) is set to `Automatically Return Credit Memo Item to Stock`. With Magento Inventory Management enabled, the item returns to the source that sent the shipment.
 
-1. Enter the quantity to refund:
+   - Update the **Qty to Refund**, and click <span class="btn">Update Qty's</span>.
 
-    - In the **Qty to Refund** box, enter the number of items to be returned. Then, press the **Enter** key to record the change.
+      ![]({% link images/images/invoice-credit-memo-items-to-refund.png %}){: .zoom}
+      _Intems to Refund_
```
Comment (lazyguru) [inline src/sales/credit-memo-create.md]:
```suggestion
      _Items to Refund_
```

File: src/sales/credit-memo-create.md [hunk 3]
```diff
@@ -1,115 +1,179 @@
...
+   <!--{% if "Default.B2B Only" contains site.edition %}-->
+   ![]({% link images/images-b2b/order-account-info-credit-memo.png %}){: .zoom}
+   <!--{% endif %}-->
+   _Create Credit Memo_
+
+1. Verify the information in the **Items to Refund** section, and if applicable, do the following:
+
+   - Select the **Return to Stock** checkbox if you want to return the returned product to inventory.
+
+      With Magento Inventory Management enabled, the inventory quantity returns to the source that sent the shipment. The product returns to stock automaticallys if [Product Stock Options]({% link configuration/catalog/inventory.md %}) is set to `Automatically Return Credit Memo Item to Stock`.
```
Comment (lazyguru) [inline src/sales/credit-memo-create.md]:
```suggestion
      With Magento Inventory Management enabled, the inventory quantity returns to the source that sent the shipment. The product returns to stock automatically if [Product Stock Options]({% link configuration/catalog/inventory.md %}) is set to `Automatically Return Credit Memo Item to Stock`.
```

**Comment by ccondra** [comment]
Thanks, Lazy Guru!  :-)

**Comment by leslietilling** [comment]
This image file name is a problem: /images/images:images:order-invoice-capture-online.png

File: src/sales/credit-memo-create.md [hunk 4]
```diff
@@ -1,115 +1,179 @@
...
 
-    <!--{% endif %}-->
-1. To add a comment to the completed credit memo, scroll down to the Comments History section and enter the comment in the box.
+1. To add a comment, enter the text in the **Credit Memo Comments** box. Then, do the following:
 
-   A history of all activity related to the order is listed.
+   - To send an email notification to the customer, mark the **Email Copy of Credit Memo** checkbox.
 
-    - To send the comment to the customer by email, mark the **Notify Customer by Email** checkbox.
+   - To include the comments you have entered in the email, mark the **Append Comments** checkbox.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
   - To include the comments you have entered in the email, select the **Append Comments** checkbox.
```

File: src/sales/credit-memo-create.md [hunk 5]
```diff
@@ -1,115 +1,179 @@
...
-    <!--{% if "Default.B2B Only" contains site.edition %}-->
-    - Refund to Company Credit
+   - If the purchase was paid with store credit, mark the **Refund to Store Credit** checkbox to credit the amount to the customer account balance.
 
-    <!--{% endif %}-->
-1. To add a comment to the completed credit memo, scroll down to the Comments History section and enter the comment in the box.
+1. To add a comment, enter the text in the **Credit Memo Comments** box. Then, do the following:
 
-   A history of all activity related to the order is listed.
+   - To send an email notification to the customer, mark the **Email Copy of Credit Memo** checkbox.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
   - To send an email notification to the customer, select the **Email Copy of Credit Memo** checkbox.
```

File: src/sales/credit-memo-create.md [hunk 6]
```diff
@@ -1,115 +1,179 @@
...
 
-    - Refund Offline
+   - In the **Adjustment Fee** field, enter a value to be subtracted from the total amount refunded.
 
-    - Refund Online
+      This amount is not subtracted from a specific section of the order such as shipping, items, or tax.
 
-    <!--{% if "Default.B2B Only" contains site.edition %}-->
-    - Refund to Company Credit
+   - If the purchase was paid with store credit, mark the **Refund to Store Credit** checkbox to credit the amount to the customer account balance.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
   - If the purchase was paid with store credit, select the **Refund to Store Credit** checkbox to credit the amount to the customer account balance.
```

File: src/sales/credit-memo-create.md [hunk 7]
```diff
@@ -1,115 +1,179 @@
...
+      _Items to Refund_
 
-    - If the purchase was paid with store credit, mark the **Refund to Store Credit** checkbox to credit the amount to the customer account balance.
+### Method 2: Issue an offline refund for check or money order
 
-    - To add a comment, enter the text in the **Credit Memo Comments** box.
+1. On the _Admin_ sidebar, click **Sales** > **Orders**.
 
-    - To send an email notification to the customer, mark the **Email Copy of Credit Memo** checkbox.
+1. Find the completed order in the grid. Then click the **View** link to open the order.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
1. Find the completed order in the grid and click the **View** link to open it.
```

File: src/sales/credit-memo-create.md [hunk 8]
```diff
@@ -1,115 +1,179 @@
...
-    ![]({% link images/images/order-create-credit-memo.png %}){: .zoom}
-    <!--{% if "Default.B2B Only" contains site.edition %}-->
-    ![]({% link images/images-b2b/order-account-info-credit-memo.png %}){: .zoom}
-    <!--{% endif %}-->
-    _Create Credit Memo_
+   - To issue an `offline` refund, go to step #6.
+   - To issue an `online` refund, continue with step #4.
 
-    The New Credit Memo page looks similar to the completed order page, with an Items to Refund section that lists each item from the invoice.
+1. Click **Invoices** in the panel on the left. Find the invoice in the grid and click **View**.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
1. Click **Invoices** in the left panel.

1. Find the invoice in the grid and click **View**.
```

File: src/sales/credit-memo-create.md [hunk 9]
```diff
@@ -1,115 +1,179 @@
...
-1. Set the return to stock option:
+1. Scroll down to the **Order Total** section of the invoice, and verify that the invoice is set to `Capture Online`. Then, click <span class="btn">Submit Invoice</span>.
 
-    {:.bs-callout-info}
-    If an online payment method was used, you cannot edit these fields.
+   ![]({% link images/images/order-invoice-capture-online.png %}){: .zoom}
+   _Capture Online_
 
-    - If the product is to be returned to inventory, mark the **Return to Stock** checkbox. With Magento Inventory Management enabled, the inventory quantity returns to the source that sent the shipment.
+1. In the button bar at the top of the invoice, click **Credit Memo**. Then, verify the information in the **Items to Refund** section. If applicable, do the following:
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
1. In the button bar at the top of the invoice, click **Credit Memo**. Then, verify the information in the **Items to Refund** section. If applicable, do the following:
```
```suggestion
1. In the button bar at the top of the invoice, click **Credit Memo**.

1. Verify the information in the **Items to Refund** section and do the following, if applicable:
```

File: src/sales/credit-memo-create.md [hunk 10]
```diff
@@ -1,115 +1,179 @@
...
-    - If the purchase was paid with store credit, mark the **Refund to Store Credit** checkbox to credit the amount to the customer account balance.
+### Method 2: Issue an offline refund for check or money order
 
-    - To add a comment, enter the text in the **Credit Memo Comments** box.
+1. On the _Admin_ sidebar, click **Sales** > **Orders**.
 
-    - To send an email notification to the customer, mark the **Email Copy of Credit Memo** checkbox.
+1. Find the completed order in the grid. Then click the **View** link to open the order.
+
+1. In the button bar at the top of the page, click **Invoice**. Scroll down to the bottom of the page and click <span class="btn">Submit Invoice"</span>.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
1. In the button bar at the top of the page, click **Invoice**. Scroll down to the bottom of the page and click <span class="btn">Submit Invoice"</span>.
```
```suggestion
1. In the button bar at the top of the page, click **Invoice**.

1. Scroll down to the bottom of the page and click <span class="btn">Submit Invoice"</span>.
```

File: src/sales/credit-memo-create.md [hunk 11]
```diff
@@ -1,115 +1,179 @@
...
+   <!--{% endif %}-->
+   <!--{% if "Default.EE Only" contains site.edition %}-->
+   ![]({% link images/images/order-create-credit-memo.png %}){: .zoom}
+   <!--{% endif %}-->
+   <!--{% if "Default.B2B Only" contains site.edition %}-->
+   ![]({% link images/images-b2b/order-account-info-credit-memo.png %}){: .zoom}
+   <!--{% endif %}-->
+   _Create Credit Memo_
+
+1. Verify the information in the **Items to Refund** section, and if applicable, do the following:
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
1. Verify the information in the **Items to Refund** section and do the following, if applicable:
```

File: src/sales/credit-memo-create.md [hunk 12]
```diff
@@ -1,115 +1,179 @@
...
+
+   - Select the **Return to Stock** checkbox if you want to return the returned product to inventory.
+
+      With Magento Inventory Management enabled, the inventory quantity returns to the source that sent the shipment. The product returns to stock automatically if [Product Stock Options]({% link configuration/catalog/inventory.md %}) is set to `Automatically Return Credit Memo Item to Stock`.
+
+   - Update the **Qty to Refund**, and click <span class="btn">Update Qty's</span>.
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
   - Update the **Qty to Refund** and click <span class="btn">Update Qty's</span>.
```

File: src/sales/credit-memo-create.md [hunk 13]
```diff
@@ -1,115 +1,179 @@
...
-    - Refund Online
+      This amount is not subtracted from a specific section of the order such as shipping, items, or tax.
 
-    <!--{% if "Default.B2B Only" contains site.edition %}-->
-    - Refund to Company Credit
+   - If the purchase was paid with store credit, mark the **Refund to Store Credit** checkbox to credit the amount to the customer account balance.
 
-    <!--{% endif %}-->
-1. To add a comment to the completed credit memo, scroll down to the Comments History section and enter the comment in the box.
+1. To add a comment, enter the text in the **Credit Memo Comments** box. Then, do the following:
```
Comment (leslietilling) [inline src/sales/credit-memo-create.md]:
```suggestion
1. To add a comment, enter the text in the **Credit Memo Comments** box and do the following:
```

**Comment by leslietilling** [comment]
This has a 2.4 label. Is it for that release?

**Comment by ccondra** [comment]
> This image file name is a problem: /images/images:images:order-invoice-capture-online.png

I don't see what's wrong with this image file name.  Your comment shows 2 colons in the path, but the source is like this:  images/images/order-invoice-capture-online.png. It looks fine in staging.

**Comment by ccondra** [comment]
running tests

**Pull request merged**